lib/hyrax/transactions/steps/apply_collection_type_permissions.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+module Hyrax
+  module Transactions
+    module Steps
+      ##
+      # A `dry-transcation` step that applies permission templates from a
+      # collection type on a given collection.
+      #
+      # @since 3.2.0
+      class ApplyCollectionTypePermissions
+        include Dry::Transaction::Operation
```

Comment:
this seems to be the deprecated transaction infrastructure

should it be:
```suggestion
        include Dry::Monads[:result]
```